vsphere-mixin/variables.libsonnet

```diff
@@ -0,0 +1,232 @@
+      local clusterLabels = this.config.clusterLabels,
+      local hostLabels = this.config.hostLabels,
+      local hostOptionalLabels = ['vcenter_cluster_name'],
+      local virtualMachineLabels = this.config.virtualMachineLabels,
+      local virtualMachineOptionalLabels = ['vcenter_cluster_name', 'vcenter_resource_pool_inventory_path', 'vcenter_virtual_app_inventory_path'],
+      local clusterLabel = 'vcenter_cluster_name',
+      local clusterSelector = 'job=~"integrations/vsphere",job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name"',
+      local hostLabel = 'vcenter_host_name',
+      local hostQuery = 'query_result(sum(vcenter_vm_memory_usage_mebibytes{job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|"}) by (vcenter_host_name))',
+      local hostRegex = '/vcenter_host_name="([^"]*)/',
+      local resourcePoolLabel = 'vcenter_resource_pool_inventory_path',
+      local resourcePoolSelector = 'job=~"integrations/vsphere",job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|",vcenter_host_name=~"$vcenter_host_name"',
+      local virtualAppLabel = 'vcenter_virtual_app_inventory_path',
+      local virtualAppSelector = 'job=~"integrations/vsphere",job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|",vcenter_host_name=~"$vcenter_host_name"',
+      local vmLabel = 'vcenter_vm_name',
+      local vmQuery = 'query_result((sum(label_join(sgn(sum by(vcenter_resource_pool_inventory_path,vcenter_vm_name)(vcenter_vm_memory_usage_mebibytes{job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|",vcenter_host_name=~"$vcenter_host_name",vcenter_resource_pool_inventory_path=~"$vcenter_resource_pool_inventory_path",vcenter_resource_pool_inventory_path!=""})),"vm_path","/","vcenter_resource_pool_inventory_path","vcenter_vm_name")) by (vm_path,vcenter_vm_name)) or (sum(label_join(sgn(sum by(vcenter_virtual_app_inventory_path,vcenter_vm_name)(vcenter_vm_memory_usage_mebibytes{job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|",vcenter_host_name=~"$vcenter_host_name",vcenter_virtual_app_inventory_path=~"$vcenter_virtual_app_inventory_path",vcenter_virtual_app_inventory_path!=""})),"vm_path","/","vcenter_virtual_app_inventory_path","vcenter_vm_name")) by (vm_path,vcenter_vm_name)) or (sum(label_replace(sgn(sum by(vcenter_resource_pool_inventory_path,vcenter_virtual_app_inventory_path,vcenter_vm_name)(vcenter_vm_memory_usage_mebibytes{job=~"$job",vcenter_datacenter_name=~"$vcenter_datacenter_name",vcenter_cluster_name=~"$vcenter_cluster_name|",vcenter_host_name=~"$vcenter_host_name",vcenter_resource_pool_inventory_path="",vcenter_virtual_app_inventory_path=""})),"vm_path","$1","vcenter_vm_name","(.*)")) by (vm_path,vcenter_vm_name)))',
```

Comment:
This looks really complex . Could explain a bit why such query is needed? Also, could you maybe try to define it using multiline syntax so it is a bit easier to read? thanks.

Reply:
Perhaps, even leaving a comment in code would be great.

Reply:
@v-zhuravlev Added a comment here for this

Reply:
I tried to split this up into a multiline, but I was having trouble with the linter because of the `query_result` I think